lib/sync/storage/dataset-clients.js

```diff
@@ -252,7 +252,9 @@ function doReadDatasetClientWithRecordsUseCache(datasetClientId, callback) {
         if (err) {
           return callback(err);
         }
-        cacheClient.set(cacheKey, JSON.stringify(results));
+        if (results) {
+          cacheClient.set(cacheKey, JSON.stringify(results));
```

Comment:
should an err be sent as first param if there are no results?
How does the caller handle a scenario where there is no error *and* no results object